### PR TITLE
RavenDB-16846 Default index deployment mode

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts [hunk 1]
```diff
@@ -195,7 +195,7 @@ class editIndex extends viewModelBase {
                         });
                     return canActivateResult;
                 } else {
-                    this.editedIndex(indexDefinition.empty());
+                    this.editedIndex(indexDefinition.empty(this.defaultDeploymentMode()));
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts]: like we talked I think we have race here.

File: src/Raven.Studio/typescript/models/database/index/indexDefinition.ts [hunk 2]
```diff
@@ -380,15 +380,15 @@ class indexDefinition {
         }
     }
     
-    static empty(): indexDefinition {
+    static empty(defaultDeploymentMode: Raven.Client.Documents.Indexes.IndexDeploymentMode): indexDefinition {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/index/indexDefinition.ts]: rename param

File: src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/indexes/editIndex.ts]: removed

File: src/Raven.Studio/typescript/models/database/index/indexDefinition.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/index/indexDefinition.ts]: removed

**Review by ml054 — APPROVED** [review]

**Pull request merged**